includes/payment-tokens/class-wc-stripe-payment-tokens.php

```
@@ -151,14 +157,152 @@ public static function customer_has_saved_methods( $customer_id ) {
+			WC_Stripe_Logger::error( 'Error getting customer payment tokens (upe) for customer: ' . $customer_id, [ 'error_message' => $e->getMessage() ] );
+		}
+
+		$tokens = $this->cleanup_invalid_tokens( $tokens, $stored_tokens, $deprecated_tokens );
+
+		// When OCS is enabled and the main stripe gateway is queried (e.g. blocks checkout), also include
+		// tokens for sub-gateways (stripe_sepa_debit, stripe_bancontact, etc.) since all payment methods
+		// are rendered under the single 'stripe' gateway in that context.
+		// The filter is kept active so each sub-gateway call goes through the normal sync path.
+		// There is no recursion risk because the OCS block only runs when gateway_id === 'stripe'.
+		if ( $gateway->is_oc_enabled() && WC_Stripe_UPE_Payment_Gateway::ID === $gateway_id ) {
+			foreach ( $this->get_sub_gateway_ids() as $sub_gateway_id ) {
```

PHPMD (2.15.0)

[error] The variable $sub_gateway_id is not named in camelCase. (undefined)

(CamelCaseVariableName)